python/ray/serve/_private/router.py

```diff
@@ -402,6 +403,41 @@ def __init__(
         self._pending_requests_to_fulfill: Deque[PendingRequest] = deque()
         self._pending_requests_to_schedule: Deque[PendingRequest] = deque()
 
+        # Prepare scheduler metrics.
+        self._actor_id: str = self._get_actor_id()
+
+        self.num_scheduling_tasks_gauge = metrics.Gauge(
+            "serve_num_scheduling_tasks",
+            description="The number of request scheduling tasks in the router.",
+            tag_keys=("app", "deployment", "actor_id"),
+        ).set_default_tags(
+            {
+                "app": self._deployment_id.app,
+                "deployment": self._deployment_id.name,
+                "actor_id": self._actor_id,
+            }
+        )
+        self.num_scheduling_tasks_gauge.set(0)
+
+        self.num_scheduling_tasks_in_backoff = 0
+        self.num_scheduling_tasks_in_backoff_gauge = metrics.Gauge(
+            "serve_num_scheduling_tasks_in_backoff",
+            description=(
+                "The number of request scheduling tasks in the router "
+                "that are undergoing backoff."
+            ),
+            tag_keys=("app", "deployment", "actor_id"),
+        ).set_default_tags(
+            {
+                "app": self._deployment_id.app,
+                "deployment": self._deployment_id.name,
+                "actor_id": self._actor_id,
```

Comment:
We can't add actor-id as a tag, this will blow up our cardinality with every actor recycle

Reply:
I see, is there a better way to distinguish between routers on different actors on the same node?

Reply:
Chatted offline:

 - We already have replica-ids tagged on other metrics
 - This metric is not being added to a dashboard (so impact should be bounded)